modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ * @return array $metadata The attachment metadata.
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get_dominant_color( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( is_bool( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ * @return mixed $attr Attributes for the image markup.
+ */
+function dominant_color_update_attachment_image_attributes( $attr, $attachment ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
```

Comment:
```suggestion
if ( !isset( $image_meta['has_transparency'] ) || !isset( $image_meta['dominant_color'] )){
    return $attr;
}
```

Might be cleaner to just exist early here. It would make the logic below a lot cleaner.

Reply:
Why was this marked as resolved?